### DISCUSSION TITLE
How do I reset the prometheus internal state?

### DISCUSSION THREAD

**Discussion by PhilipMay** [prometheus/client_python#828] [Q&A]
I have a module that uses prometheus to expose some metrics.
Then I have a pytest module that tests that first module with mutiple test functions.

Between the test functions the state of the prometheus client is not reset.
But I want to have a "clean and new" state for every single test function.
As the state was when the Python runtime started.

My question: How can I clean / reset / delete the state of the prometheus client?

Many thanks
Philip

**Comment by csmarchbanks** [comment]
Ideally you would be able to use a custom registry in your module rather than the default registry, e.g. `registry=my_registry`. This can also be helpful for users of your module who may not want all of the default collectors.

There is also an `unregister` function on the registry that you can use to iterate across all of the registered metrics that you would like to reset and unregister them.
